### DISCUSSION TITLE
SSVM connectivity issue

### DISCUSSION THREAD

**Discussion by tampler** [apache/cloudstack#9755] [Q&A]
##### ISSUE TYPE

 * Improvement Request
 * Enhancement Request
 * Other

##### COMPONENT NAME

~~~
Core, Installation
~~~

##### CLOUDSTACK VERSION

~~~
4.20.0.0-snapshot.20240928 
~~~

##### CONFIGURATION

Core, Advanced Zone, NFS for Primary and Secondary storage

##### OS / ENVIRONMENT

Ubuntu 24.04

##### SUMMARY

SSVM and CPVM fail to connect to the management server :red_circle: .  Looks like a routing issue :earth_americas: 

My `mgmt` server sits on `192.168.1.1`, but SSVM tries to connect thru the `public IP` on another interface :roll_eyes: 

How do I change this in SSVM ?

```sh
> cat /var/log/cloud.log

2024-09-28T21:38:37,023 INFO  [cloud.agent.Agent] (main:[]) Connecting to host: 172.16.84.33
2024-09-28T21:38:37,023 INFO  [utils.nio.NioClient] (main:[]) Connecting to 172.16.84.33:8250
2024-09-28T21:38:40,085 ERROR [utils.nio.NioClient] (main:[]) Unable to initialize the threads. java.net.NoRouteToHostException: No route to host
```

Here's my routing table :eyes: 

```sh
> ip route

default via 172.16.84.1 dev eth2 
10.0.0.0/8 via 192.168.10.1 dev eth1 
169.254.0.0/16 dev eth0 proto kernel scope link src 169.254.219.213 
172.16.0.0/12 via 192.168.10.1 dev eth1 
172.16.84.33/28 dev eth2 proto kernel scope link src 172.16.84.1 
172.16.84.33 via 192.168.10.1 dev eth1 
192.168.0.0/16 via 192.168.10.1 dev eth1 
192.168.10.0/24 dev eth1 proto kernel scope link src 192.168.10.30 
193.106.92.92 via 192.168.10.1 dev eth1 
193.106.93.93 via 192.168.10.1 dev eth1 
```

```sh
> ip a s 

2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 0e:00:a9:fe:db:d5 brd ff:ff:ff:ff:ff:ff
    altname enp0s3
    altname ens3
    inet 169.254.219.213/16 brd 169.254.255.255 scope global eth0
       valid_lft forever preferred_lft forever

3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 1e:00:fe:00:00:1d brd ff:ff:ff:ff:ff:ff
    altname enp0s4
    altname ens4
    inet 192.168.10.30/24 brd 192.168.10.255 scope global eth1
       valid_lft forever preferred_lft forever

4: eth2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 1e:00:b1:00:00:32 brd ff:ff:ff:ff:ff:ff
    altname enp0s5
    altname ens5
    inet 172.16.84.33/28 brd 172.16.84.14 scope global eth2
       valid_lft forever preferred_lft forever
```

How and where do I update the management server address so that my SSVM can connect to the mgmt server :roll_eyes: ?

I need SSVM to connect to my `mgmt`  via `eth1` on `192.168.1.1`.

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
SSVM and CPVM connects to the mgmt server
~~~

##### ACTUAL RESULTS

~~~
SSVM and CPVM can't connect to the mgmt server
~~~

**Comment by weizhouapache** [comment]
You can check global configuration "host"

**Comment by rajujith** [comment]
@tampler  can you check the global configuration "management.network.cidr" as well.

**Reply by tampler** [reply]
> @tampler can you check the global configuration "management.network.cidr" as well.

It's empty by default in ACS 20.0.0. When I set it to `192.168.1.1/24`, `cloudstack-ssh` stops working at all :roll_eyes:

**Comment by DaanHoogland** [comment]
@tampler , have you found a resolution to your query yet?

**Reply by tampler** [reply]
@DaanHoogland 

Hi Dan

I couldn't fix this and was annoyed due to constant installation issues :roll_eyes: . I've been using ACS since 2018 till 2024. The same installation issue with SSMV and storage occurred all the time :disappointed: 

So I dropped ACS usage in favour of other technologies and stacks so can't test/debug/validate this issue anymore.

Perhaps, I'll be back to ACS in future, but now I can't contribute and provide any feedback. Sorry :pray:
